aixplain/utils/cache_utils.py

```
@@ -2,26 +2,35 @@
 import json
 import time
 import logging
+from filelock import FileLock
 
-CACHE_DURATION = 24 * 60 * 60
-CACHE_FOLDER = ".aixplain_cache"
+CACHE_FOLDER = ".cache"
+CACHE_FILE = f"{CACHE_FOLDER}/cache.json"
+LOCK_FILE = f"{CACHE_FILE}.lock"
+DEFAULT_CACHE_EXPIRY = 86400
 
 
-def save_to_cache(cache_file, data):
+def get_cache_expiry():
+    return int(os.getenv("CACHE_EXPIRY_TIME", DEFAULT_CACHE_EXPIRY))
+
+
+def save_to_cache(cache_file, data, lock_file):
     try:
         os.makedirs(os.path.dirname(cache_file), exist_ok=True)
-        with open(cache_file, "w") as f:
-            json.dump({"timestamp": time.time(), "data": data}, f)
+        with FileLock(lock_file):
+            with open(cache_file, "w") as f:
+                json.dump({"timestamp": time.time(), "data": data}, f)
     except Exception as e:
         logging.error(f"Failed to save cache to {cache_file}: {e}")
```

why we're hiding the exception and log it? If we have no valid plan or purpose for preventing an exception to raise, we should let it raise to halt the program to proceed with an unwanted data flow.